Add Ionic Capacitor

lubej opened the pull request:
Folders android and ios were auto generated with npx cap add [platform]. [Android platform](https://capacitorjs.com/docs/android#adding-the-android-platform) - example for Android.

```
npx cap add android
npx cap add ios
```

Assets for each platform were generated using [@capacitor/assets](https://github.com/ionic-team/capacitor-assets#usage---custom-mode)

buberdds commented:
Before adding all these MBs to repo we should run squoosh or some other optimizer. 
sample splash png: 2.6MB -> (98% down) 38kB

buberdds commented on package.json:
```diff
@@ -57,7 +59,9 @@
     "@capacitor-community/bluetooth-le": "3.0.0",
+    "@capacitor/android": "5.0.4",
     "@capacitor/core": "5.0.4",
+    "@capacitor/ios": "5.0.4",
```
can you bump to 5.6.0 ? would be cool to update core as well.

lubej replied:
No, I expect before this gets merged, there will be at least 10 new version. I know this version works with current version of https://github.com/oasisprotocol/ionic-ledger-hw-transport-ble. And currently in this PR there is no real way to test on real devices, as Bluetooth UI is still missing - I was planning to add it after this PR is merged.

buberdds commented on android/gradlew.bat:
```diff
@@ -0,0 +1,92 @@
+@rem
```
Do we need files for Win?

lubej replied:
I wouldn't touch any of autogenerated scripts. Just to spend ages fixing it, if something breaks in the future.

buberdds commented on android/gradle/wrapper/gradle-wrapper.jar:
no idea what this compiled file contains, also no idea if that's common to store compiled Java in repo.

lubej replied:
I would say that it is there for a reason. And I wouldn't touch autogenerated files.

buberdds commented on android/app/src/test/java/com/getcapacitor/myapp/ExampleUnitTest.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Test
+    public void addition_isCorrect() throws Exception {
+        assertEquals(4, 2 + 2);
```
good that we have some test coverage here :trollface:

lubej replied:
I would still rather keep this, just in case there is no confusion if we need to generate android/ios platform from scratch, and wonder where did this file came from disappear. But if you are strongly against it, I can delete it as well.

buberdds commented on ios/App/Podfile.lock:
```diff
@@ -0,0 +1,28 @@
+
+PODFILE CHECKSUM: 47998e176d73fb1dcd46c2cd81d1fab8a9f392d8
+
+COCOAPODS: 1.12.1
```
Do we have readme with requirements (like cacaopods on Mac etc) how to setup and build all these files ?

lubej replied:
`ionic cordova build ios` all taken care of by Capacitor

buberdds commented on android/app/proguard-rules.pro:
```diff
@@ -0,0 +1,21 @@
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# If your project uses WebView with JS, uncomment the following
```
should we uncomment something here?

lubej replied:
No

buberdds commented:
Can you provide steps how these files were generated - sorry I did not look into ionic since weeks and I remember nothing. thx

lubej commented:
> Can you provide requirements and steps how these files were generated - sorry I did not look into ionic since weeks and I remember nothing. thx

So all files inside `android` and `ios` folders are autogenerated. 

`ionic cordova platform add android` 
`ionic cordova platform rm ios`

And I wouldn't change those files, I know there is some useless stuff in it - addressing some comments in the PR.

lukaw3d commented:
What was the reason for including these autogenerated files into the repo again? Is it because of unavoidable modifications we need to make to them later on?

lubej commented:
> What was the reason for including these autogenerated files into the repo again? Is it because of unavoidable modifications we need to make to them later on?

Yeah. Even though we could in theory just commit diffs, and overwrite generated files on `postinstall` script. But i am not sure if different version of ionic will generate the same files.
